### PR TITLE
Add synthetic PIRB reviewer testing surface

### PR DESCRIPTION
**Pull request by JustAGhosT**
## Summary

- add an admin-only Domain Reviewer Lab for synthetic PIRB workflow rehearsal
- enforce a strict versioned contract with fixed fictional scenarios, enumerated findings, and no free-text inputs
- fail closed on candidate/credential/household data and prohibit persistence, registry calls, outreach, payment, recording, posting, and production access
- keep PIRB eligibility unevaluated and O5 inactive
- add API, contract, component, navigation tests, and the provider-boundary documentation

## Why

PIRB-aligned domain review needs a safe testing surface before any live provider verification integration. This gives administrators a deterministic rehearsal environment without handling real candidate data or causing external effects.

## Validation

- `pnpm exec vitest run tests/lib/domain-safety-trial.test.ts tests/api/domain-safety-reviewer-trial.test.ts tests/components/domain-reviewer-lab-page.test.tsx tests/lib/nav-config.test.ts` — 13/13 passed
- `pnpm run lint` — passed
- `pnpm exec tsc --noEmit` — passed serially after build route generation
- `pnpm run build` — passed
- Prettier changed-file check and `git diff --check` — passed
- full `pnpm test` — 392/394; only the two existing Windows CRLF deployment-workflow contract failures remain
- browser: Hans/admin completed Variant B to `ready_for_internal_replay`; Charl/operator was redirected and received API 403

## Safety boundary

This PR does not call PIRB, store trial records, process candidate PII, contact reviewers, issue invitations, record sessions, pay participants, post results, or activate O5. A live/manual provider adapter remains gated on the O6 restricted-store and privacy prerequisites.

Baton: `1d10b5fd-1b1b-4216-ada5-80942663fc83`

### CODE REVIEW

File: lib/reviewer-trials/domain-safety-trial.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+): DomainSafetyTrialEvaluation {
+  const criticalFailures = DOMAIN_SAFETY_CRITICAL_GATES.filter(
+    ({ id }) => submission.criticalGates[id] === "fail"
+  ).map(({ id }) => id)
+  const incompleteCriticalGates = DOMAIN_SAFETY_CRITICAL_GATES.filter(
+    ({ id }) => submission.criticalGates[id] === "not_tested"
+  ).map(({ id }) => id)
+  const qualityFailures = DOMAIN_SAFETY_QUALITY_DIMENSIONS.filter(
+    ({ id }) => submission.qualityDimensions[id] === "failure"
+  ).map(({ id }) => id)
```
[the remark this answers is not in the document]
Comment (JustAGhosT) [reply lib/reviewer-trials/domain-safety-trial.ts:251]: Implemented: the evaluator now tracks incomplete quality dimensions and returns close_without_reliance until every quality dimension is exercised. Added a regression test.

File: app/dashboard/hans/reviewer-lab/page.tsx [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+            <h1 className="text-foreground flex items-center gap-3 text-2xl font-bold sm:text-3xl">
+              <FlaskConical className="text-primary h-8 w-8" />
+              Domain Reviewer Lab
+            </h1>
+            <p className="text-muted-foreground mt-1 max-w-3xl">
+              Rehearse the synthetic South African plumbing-review contract before any PIRB
+              verification, candidate contact, or Gate activation.
+            </p>
+          </div>
+          <Button variant="outline" onClick={() => void load()} disabled={loading}>
```
[the remark this answers is not in the document]
Comment (JustAGhosT) [reply app/dashboard/hans/reviewer-lab/page.tsx:207]: Implemented: Reset rehearsal now clears the definition, variant, gates, quality results, finding state, acknowledgements, submission state, errors, and stale evaluation before reloading. Added a component regression test.
